ypathgen/pathgen_test.go

```diff
@@ -867,6 +868,43 @@ func TestGeneratePathCode(t *testing.T) {
 		inSchemaStructPkgPath:                  "",
 		inPathStructSuffix:                     "Path",
 		wantStructsCodeFile:                    filepath.Join(TestRoot, "testdata/structs/openconfig-camelcase.path-txt"),
+	}, {
+		name:                       "openconfig import exclude test",
+		inFiles:                    []string{filepath.Join(datapath, "openconfig-import.yang")},
+		inIncludePaths:             []string{datapath},
+		wantStructsCodeFile:        filepath.Join(TestRoot, "testdata/structs/openconfig-import.path-txt"),
+		inPreferOperationalState:   true,
+		inShortenEnumLeafNames:     true,
+		inGenerateWildcardPaths:    true,
+		inExcludeSearchPathModules: true,
+		inSchemaStructPkgPath:      "",
+		inPathStructSuffix:         "Path",
+		checkYANGPath:              true,
+		wantNodeDataMap: NodeDataMap{
+			"ChildPath": {
+				GoTypeName:            "*Child",
+				LocalGoTypeName:       "*Child",
+				GoFieldName:           "Child",
+				SubsumingGoStructName: "Child",
+				YANGPath:              "/openconfig-import/child",
+			},
+			"Child_TestPath": {
+				GoTypeName:            "string",
+				LocalGoTypeName:       "string",
+				GoFieldName:           "Test",
+				SubsumingGoStructName: "Child",
+				IsLeaf:                true,
+				IsScalarField:         true,
+				YANGTypeName:          "string",
+				YANGPath:              "/openconfig-import/child/test",
+			},
+			"DevicePath": {
+				GoTypeName:            "*Device",
+				LocalGoTypeName:       "*Device",
+				SubsumingGoStructName: "Device",
+				YANGPath:              "/",
+			},
+		},
```

Comment:
Thanks. Could you add a test that verifies the opposite behaviour when the flag is false? To be safe you should add it for both GoStruct and PathStruct generation generation. You don't have to test this field if it's taking too long though, feel free to assign a TODO for me: https://github.com/openconfig/ygot/issues/580.

Reply:
added changed import to openconfig-simple-target which simper. kept nodedatamap for now